src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -582,6 +602,15 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             }
         }
 
+        private bool TimeoutFired(CancellationToken callerToken, long timeoutTime) => !callerToken.IsCancellationRequested && timeoutTime > 0 && Environment.TickCount64 >= timeoutTime;
```

Comment:
If you initialize timeoutTime to be long.MaxValue instead of 0, you can remove the `timeoutTime > 0` check.  It's also more consistent, essentially saying the timeout is so far in the future it's infinite.

Reply:
Done